### PR TITLE
Rename CLI to sb

### PR DESCRIPTION
**Pull request by shilman**
Issue: #4168 

## What I did

Per discussion with @Keraito @tmeasday @ndelangen

1. Rename storybook CLI to `sb` to avoid conflicts with storybook NPM script. We use `yarn storybook` everywhere and creating a CLI called `storybook` introduces a name conflict. After the rename, users can add the CLI locally in the project and access it by running `yarn sb xxx`. This isn't important now since `init` is the only supported command, but it will be important later when we start adding more commands. Plus `sb` is shorter.
2. Remove start/build commands. `sb start/build` should call start-storybook and build-storybook directly, rather than calling `yarn storybook`. This way we can use `sb start/build` in the `package.json` NPM scripts and expose users to the CLI that way. For now let's remove it and keep things simple.
3. Update documentation

## Note

I tried to make `npx @storybook/cli init` work, but it currently uses `getstorybook` as the associated binary. I reordered the binary declaration in this PR, but that didn't seem to change things. For now `-p` works and is much better than a global install IMHO.

## How to test

```
npx -p @storybook/cli@test sb init
```

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -134,6 +135,14 @@ Also make sure you have a `.babelrc` in your project directory. You probably alr
 
 If you're using `start-storybook` on CI, you may need to opt out of this using the new `--ci` flag.
 
+### CLI Rename
+
+We've deprecated the `getstorybook` CLI in 4.0. The new way to install storybook is `sb init`. We recommend installing using `npx` for convenience and to make sure you're always using the latest version of the CLI:
```
Comment (Keraito) [inline MIGRATION.md]: ~installing~

File: lib/cli/lib/initiate.js [hunk 2]
```diff
@@ -179,10 +179,10 @@ export default function(options, pkg) {
       default:
         paddedLog(`We couldn't detect your project type. (code: ${projectType})`);
         paddedLog(
-          "Please make sure you are running the `storybook init` command in your project's root directory."
+          "Please make sure you are running the `sb init` command in your project's root directory."
         );
         paddedLog(
-          'You can also install storybook for plain HTML snippets with `storybook init --html` or follow some of the slow start guides: https://storybook.js.org/basics/slow-start-guide/'
+          'You can also install storybook for plain HTML snippets with `sb init --html` or follow some of the slow start guides: https://storybook.js.org/basics/slow-start-guide/'
```
Comment (Keraito) [inline lib/cli/lib/initiate.js]: These two logs were [replaced](https://github.com/storybooks/storybook/pull/4184/files#diff-3e4b1a44a581c8cf5952fefaf486682dR162) in #4184. Just a heads up depending on which PR gets merged first.

**Review by Keraito — APPROVED** [review]
Small typo. LGTM

File: lib/cli/lib/initiate.js (hunk 2 above)
Comment (shilman) [reply lib/cli/lib/initiate.js]: @Keraito Merging now --- sorry to create extra work on the resolve, but my change is a small one

Comment (Keraito) [reply lib/cli/lib/initiate.js]: No problem, won't be too much

**Pull request merged**